control-plane/agents/src/bin/core/pool/registry.rs

```diff
@@ -3,10 +3,37 @@ use crate::controller::{
     wrapper::{GetterOps, *},
 };
 use agents::errors::{SvcError, SvcError::PoolNotFound};
-use stor_port::types::v0::transport::{CtrlPoolState, NodeId, Pool, PoolId, Replica, ReplicaId};
+use stor_port::types::v0::transport::{
+    CtrlPoolState, NodeId, Pool, PoolId, Replica, ReplicaId, VolumeId,
+};
 
 /// Pool helpers
 impl Registry {
+    /// Get all pools.
+    pub(crate) async fn get_volume_opts_pools(
+        &self,
+        volume_uuid: VolumeId,
+    ) -> Result<Vec<Pool>, SvcError> {
+        let mut pools: Vec<Pool> = vec![];
+        match self.volume(&volume_uuid).await {
```

Comment:
I don't think you need to do this.
You can just get the volume replicas, and map their pool ids.
In fact, `volume_data_nodes` already does this same thing, so you can refactor that function, add a new function which returns a vec of pool ids.

Reply:
ah, thanks  this was quiet easy.